### ISSUE TITLE
No compileJava command

### ISSUE THREAD

**Issue by IgorGanapolsky** [tbroyer/gradle-errorprone-plugin-v0.0.x#22]
> - What went wrong:
>   Task 'compileJava' not found in root project 'hello-libs'.

I keep getting this error when I try to run your instructions.  Please advise.

Igor

**Comment by tbroyer** [comment]
If you don't have a `compileJava` task, then that means you don't have the java plugin applied: https://docs.gradle.org/current/userguide/java_plugin.html#N15056

**Comment by IgorGanapolsky** [comment]
Did you specify that in your README?

**Comment by tbroyer** [comment]
No, I'm assuming people coming to this plugin know about what errorprone is, and know slightly enough of Gradle to understand what they're doing ([`JavaCompile` tasks](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.compile.JavaCompile.html), etc.)
That was before the rise of the new Gradle-based Android build though.
I don't do any Android development so I can't really give a compelling example; all I know is that Android builds are awfully complex, so you'd better not go the "advanced usage" route if you're not first an "advanced Gradle user".

**Comment by IgorGanapolsky** [comment]
Yes, I am an Android gradle user, and applying the `JavaCompile` task is
foreign to me.  I think Google should've warned that errorprone isn't
really for beginner gradle Android users...

**Comment by chalup** [comment]
You probably need to do something like this:

``` groovy
import net.ltgt.gradle.errorprone.ErrorProneToolChain

apply plugin: 'com.android.application'
apply plugin: 'net.ltgt.errorprone-base'

afterEvaluate {
  android.applicationVariants.each { it.javaCompile.toolChain ErrorProneToolChain.create(project) }
}
```

**Comment by IgorGanapolsky** [comment]
This **afterEvaluate** doesn't work in the Gradle experimental version, which is based on the *_model { } *_ closure.

**Comment by tbroyer** [comment]
What I don't understand is why you'd want to use the "advanced usage" for an Android build (without a specific need for it): simple usage works OK out-of-the-box; see https://github.com/artem-zinnatullin/qualitymatters/commit/bb0faa20ad1c70a429dedca4233982f3c5bdc7bd#diff-c197962302397baf3a4cc36463dce5ea

I also just applied the same changes to https://github.com/googlesamples/android-ndk/tree/master/hello-libs (which I think is the one you wanted to build) and it works out-of-the-box too:

``` diff
diff --git a/hello-libs/app/build.gradle b/hello-libs/app/build.gradle
index 1707001..b7ca626 100644
--- a/hello-libs/app/build.gradle
+++ b/hello-libs/app/build.gradle
@@ -1,4 +1,5 @@
 apply plugin: 'com.android.model.application'
+apply plugin: "net.ltgt.errorprone"

 // Root of 3rd party lib(s): location could be anywhere on the host system
 def lib_distribution_root = '../distribution'
diff --git a/hello-libs/build.gradle b/hello-libs/build.gradle
index ccfbb5c..033c05b 100644
--- a/hello-libs/build.gradle
+++ b/hello-libs/build.gradle
@@ -3,12 +3,16 @@
 buildscript {
     repositories {
         jcenter()
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
     }
     dependencies {
         classpath 'com.android.tools.build:gradle-experimental:0.7.0'

         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
+        classpath "net.ltgt.gradle:gradle-errorprone-plugin:0.0.8"
     }
 }
```

To test whether the plugin is correctly applied, run Gradle with `--debug` and grep for `ErrorProneCompiler` or `Compiling with error-prone compiler`.

**Comment by IgorGanapolsky** [comment]
@tbroyer Thanks for the instructions and this nice plugin.  @changetip $1.

**Comment by changetip** [comment]
Hi @tbroyer, @IgorGanapolsky sent you a tip worth 1,485 bits ($1.00), and I'm here to deliver it ➔ **[collect your tip](https://www.changetip.com/collect/948365)**.

**[Learn more about ChangeTip](https://www.changetip.com/tip-online/github)**

**Comment by stephanenicolas** [comment]
Indeed the problem is not so much about deprecation, it's that by using only the old task creation API, the error prone plugin, especially on android, is forcing the creation of all compilation tasks for all variants during the configuration phase. 

Thx for adressing this @tbroyer

**Comment by tbroyer** [comment]
I finally decided to only support AGP 3.3 (in 65b1026ebeae1b7ed8c28578c7f6eea512c16bea) and break compatibility with previous versions (it turns out that v0.6.1 is compatible with Gradle 5.2 so AGP 3.2 users can stay on v0.6.1). I've just released v0.7 with the change.
